### PR TITLE
refactor: lint and style updates

### PR DESCRIPTION
**Pull request by wyardley**
#### Description
- update golang-ci in Circlel config to latest
- update pinned golangci-lint-version to match version in CI
- wrap errors
- use builtin errors (refactor by @Charles546)
- switch from a = a * b to a *= b and i = i + 1 to i++
- disable exhaustivestruct
- other minor formatting / stylle changes

#### This PR fixes the following issues
Closes #336

### CODE REVIEW

File: pkg/dipper/errorHandling.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ func (e Error) Error() string {
 	return string(e)
 }
 
-// Is check if a given error matches this error.
+// Is : check whether the given error matches this error.
```
Comment (wyardley) [inline pkg/dipper/errorHandling.go:19]: question (maybe a dumb one) - does this do something different from `errors.Is` from the builtin errors package?

File: pkg/dipper/errorHandling.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ func (e Error) Error() string {
...
 func (e Error) Is(target error) bool {
 	if t, ok := target.(Error); ok {
```
Comment (wyardley) [inline pkg/dipper/errorHandling.go:21]: This is the one remaining error (aside from tests, which are ignored for now):

```
pkg/dipper/errorHandling.go:21:14: type assertion on error will fail on wrapped errors. Use errors.As to check for specific errors (errorlint)
	if t, ok := target.(Error); ok {
	            ^
```

**Comment by Charles546** [comment]
@wyardley I PR'ed your branch. https://github.com/wyardley/honeydipper/pull/2

**Review by cynful — APPROVED** [review]
nice

**Pull request merged**